app/src/androidTest/java/sw19/moring03/paint/RedoButtonTest.java

```diff
@@ -0,0 +1,109 @@
+package sw19.moring03.paint;
+
+import android.graphics.drawable.Drawable;
+import android.support.test.espresso.Espresso;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v7.view.menu.ActionMenuItemView;
+import android.support.v7.widget.Toolbar;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+
+import sw19.moring03.paint.Views.DrawingView;
+import sw19.moring03.paint.tools.Tools;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.swipeLeft;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static sw19.moring03.paint.util.Interaction.changeValueOfStrokeWidthSeekBar;
+import static sw19.moring03.paint.util.Interaction.touchAt;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class RedoButtonTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> launchActivityRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Test
+    public void testButtonVisibility() {
```

Comment:
pls add a undo redo undo test or something like that. check if all the lists have the correct amount of elements after each of those actions